xlators/mgmt/glusterd/src/glusterd-sm.c

```diff
@@ -1595,7 +1595,9 @@ glusterd_friend_sm()
          *
          * */
         synclock_unlock(&priv->big_lock);
-        glusterd_launch_synctask(glusterd_spawn_daemons, NULL);
+        ret = glusterd_launch_synctask(glusterd_spawn_daemons, NULL);
+        if (ret)
+            return ret;
```

Comment:
I would say we can't return without taking the `priv->big_lock` again.

Reply:
But the task creation failed, so do we still need to take a lock? Is it because we don't check for `glusterd_friend_sm` return value anywhere in the caller?

Reply:
The function is called with the lock taken and must return with it still taken.

Reply:
Ok. I think we can avoid changes to `glusterd_friend_sm` in this patch, as the return value from this function is not being stored in the caller in any of the places. But, if we want to exit in case the thread creation fails we would need the `return` value is what I think, and making that change will make this patch big and complicated. What do you suggest @xhernandez , @mohit84 ?

Reply:
I would modify `glusterd_launch_synctask()` so that it returns an error, but in this patch I would only check the result in `glusterd_compare_friend_data()`. Other callers can be modified in other patches if necessary.